programs/clearing_house/src/controller/orders.rs

```diff
@@ -1841,3 +1937,1280 @@ pub fn pay_keeper_flat_reward(
+            initial_margin_ratio,
+            maintenance_margin_ratio,
+            None,
+        )?
+    };
+
+    if breaches_oracle_price_limits {
+        let filler_reward = {
+            let mut quote_bank = bank_map.get_quote_asset_bank_mut()?;
+            pay_keeper_flat_reward_for_spot(
+                &mut maker,
+                filler.as_deref_mut(),
+                &mut quote_bank,
+                filler_reward,
+            )?
+        };
+
+        cancel_order(
+            maker_order_index,
+            maker.deref_mut(),
+            &maker_key,
+            market_map,
+            bank_map,
+            oracle_map,
+            now,
+            slot,
+            OrderActionExplanation::OraclePriceBreachedLimitPrice,
+            Some(filler_key),
+            filler_reward,
+            false,
+        )?;
+
+        return Ok((None, None, None, None));
+    }
+
+    Ok((
+        Some(maker),
+        Some(maker_stats),
+        Some(maker_key),
+        Some(maker_order_index),
+    ))
+}
+
+fn fulfill_spot_order(
+    user: &mut User,
+    user_order_index: usize,
+    user_key: &Pubkey,
+    user_stats: &mut UserStats,
+    maker: &mut Option<&mut User>,
+    maker_stats: &mut Option<&mut UserStats>,
+    maker_order_index: Option<usize>,
+    maker_key: Option<&Pubkey>,
+    filler: &mut Option<&mut User>,
+    filler_key: &Pubkey,
+    filler_stats: &mut Option<&mut UserStats>,
+    bank_map: &BankMap,
+    market_map: &MarketMap,
+    oracle_map: &mut OracleMap,
+    now: i64,
+    slot: u64,
+    fee_structure: &FeeStructure,
+    mut serum_fulfillment_params: Option<SerumFulfillmentParams>,
+) -> ClearingHouseResult<(u128, bool)> {
+    let free_collateral = calculate_free_collateral(user, market_map, bank_map, oracle_map)?;
+
+    let base_bank_index = user.orders[user_order_index].market_index;
+    let bank_balance_index = user.get_bank_balance_index(base_bank_index)?;
+    let token_amount = user.bank_balances[bank_balance_index]
+        .get_token_amount(bank_map.get_ref(&base_bank_index)?.deref())?;
+    let bank_balance_type: BankBalanceType = user.bank_balances[bank_balance_index].balance_type;
+
+    let risk_decreasing = is_spot_order_risk_decreasing(
+        &user.orders[user_order_index],
+        &bank_balance_type,
+        token_amount,
+    )?;
+
+    if free_collateral < 0 && !risk_decreasing {
+        let filler_reward = {
+            let mut quote_bank = bank_map.get_quote_asset_bank_mut()?;
+            pay_keeper_flat_reward_for_spot(
+                user,
+                filler.as_deref_mut(),
+                &mut quote_bank,
+                fee_structure.cancel_order_fee,
+            )?
+        };
+
+        cancel_order(
+            user_order_index,
+            user,
+            &user_key,
+            market_map,
+            bank_map,
+            oracle_map,
+            now,
+            slot,
+            OrderActionExplanation::MarketOrderFilledToLimitPrice,
```

Comment:
wrong OrderActionExplanation? isnt this cancel since they at max leverage?

Reply:
yes fixing